Feat/ Translate guides/release/components/built-in-components.md (#158)
DazzlingFugu/ember-fr-guides-source: guides/release/components/built-in-components.md

Ember fournit deux composants prêts à l'emploi pour construire des formulaires:

* [`<Input>`](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/Input?anchor=Input)
* [`<Textarea>`](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/Textarea?anchor=Textarea)

> Le markup HTML de ces composants est similaire à celui des éléments `<input>` et `<textarea>`. Au contraire de ces éléments natifs, `<Input>` et `<Textarea>` mettent automatiquement à jour l'état des valeurs qui leur sont liées.

Reviewer: You forgot the "native" information
```suggestion
Le markup HTML de ces composants est similaire à celui des éléments natifs `<input>` et `<textarea>`. À la différence des éléments natifs, `<Input>` et `<Textarea>` mettent automatiquement à jour l'état des valeurs qui leur sont liées.
```

## `<Input>`

> Nous avons dit plus haut que les composants _built-in_ (c'est-à-dire "construits à l'intérieur du framework") ont un markup HTML similaire à celui de leur équivalent natif. Que'est-ce que ça signifie ?

Reviewer 2: Do we say 'la framework' or 'le framework' 🤔 I always said 'la'

Author: I have always said "le", it's rather "le" on the French Internet

> Nous avons dit plus haut que les composants _built-in_ (c'est-à-dire "construits à l'intérieur du framework") ont un markup HTML similaire à celui de leur équivalent natif. Que'est-ce que ça signifie ?

Reviewer 2: Qu'est-ce que ça signifie ? - Mistyping of _Qu'est-ce_

Author: ```suggestion
Nous avons dit plus haut que les composants _built-in_ (c'est-à-dire "construits à l'intérieur du framework") ont un markup HTML similaire à celui de leur équivalent natif. Qu'est-ce que ça signifie ?
```

Considérons l'exemple suivant, le template d'un fichier.

```handlebars
<label for="user-question">Posez une question sur Ember :</label>
<Input
  id="user-question"
  @type="text"

>   @value="Comment fonctionnent les champs textes ?"

Reviewer 2: Champs de textes?

Author: It's ok to write "champ texte", but I shouldn't have put a "s" at the end of "texte", because it's not an adjective, it's a shortcut to "de type texte"  

```suggestion
  @value="Comment fonctionnent les champs texte ?"
```

/>
```

Quand Ember rend ce template, vous pouvez voir le code HTML suivant : 

```html
<label for="user-question">Posez une question sur Ember :</label>
<input id="user-question" type="text" value="Comment fonctionnent les champs textes ?" />
```

### Associer _labels_ et _inputs_

> Chaque _input_ (champ texte) doit être associé à un _label_ (étiquette). Le composant built-in `<Input>` permet de :

Reviewer: > In HTML, there are a few ways to do this.

I think we should keep this, as it "explains" why we have a list after ^^

Author: ```suggestion
Chaque _input_ (champ texte) doit être associé à un _label_ (étiquette). Il y a différentes manières d'écrire cette association en HTML. Le composant built-in `<Input>` permet de :
```

> Chaque _input_ (champ texte) doit être associé à un _label_ (étiquette). Il y a différentes manières d'écrire cette association en HTML. Le composant built-in `<Input>` permet de :

Reviewer: ```suggestion
Chaque _input_ (champ texte) doit être associé à un _label_ (étiquette). Il y a différentes manières d'écrire cette association en HTML. Avec composant built-in `<Input>`, il est possible de :
```
I think it work better with the following text

Author: ```suggestion
Chaque _input_ (champ texte) doit être associé à un _label_ (étiquette). Il y a différentes manières d'écrire cette association en HTML. Avec le composant built-in `<Input>`, il est possible de :
```

   />
   ```

3. Utiliser l'attribut `aria-label` pour décrire l'input à l'aide d'un texte invisible à l'écran mais détectable par les technologies d'assistance.

   ```handlebars
   <Input
     aria-label="Posez une question sur Ember"
     @type="text"
     @value={{this.userQuestion}}
   />
   ```

Il est plus approprié d'utiliser l'élément `<label>`, mais l'attribut `aria-label` devient utile dans les cas où un intitulé visible n'est pas possible.

### Définir les attributs de `<Input>`

À quelques exceptions près, vous pouvez passer [les attributs de input](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#Attributes) au composant `<Input>`.

Par exemple, l'attribut `aria-labelledby` peut être utile pour implémenter un champ de recherche. Le bouton de recherche peut servir de label pour l'élément input :

```handlebars
<Input aria-labelledby="button-search" />
<button id="button-search" type="button">Rechercher</button>
```

Quand un attribut est assigné avec des doubles quotes (`"button-search"` dans l'exemple ci-dessus), sa valeur est assignée directement à l'élément.

> Vous pouvez aussi lier l'attribut `value` à une propriété définie dans votre application.

Reviewer: It's more 
```suggestion
Vous pouvez aussi lier la valeur d'un attribut à une propriété définie dans votre application.
```

/>
```

> Rappelez-vous qu'il y a quelques exceptions. Les attributs d'input suivants doivent être passés au composant `<Input>` comme argumens (les arguments sont préfixés du symbole `@`) :

Reviewer: Small typo ^^

```suggestion
Rappelez-vous qu'il y a quelques exceptions. Les attributs d'input suivants doivent être passés au composant `<Input>` comme arguments (les arguments sont préfixés du symbole `@`) :
```

### Actions

> À partir de Ember Octane, il est recommandé d'utiliser le modifier `{{on}}` pour appeler une action lors d'un _event_ (événement se produisant sur la page) comment [l'event "input"](https://developer.mozilla.org/fr/docs/Web/API/HTMLElement/input_event).

Reviewer: ```suggestion
À partir de Ember Octane, il est recommandé d'utiliser le modifier `{{on}}` pour appeler une action lors d'un _event_ (événement se produisant sur la page) comme [l'event "input"](https://developer.mozilla.org/fr/docs/Web/API/HTMLElement/input_event).
```

```handlebars
<label for="input-name">Nom :</label>
<Input
  id="input-name"
  @value={{this.name}}
  {{on "input" this.validateName}}
/>
```

[À propos du modifier `{{on}}`](../../upgrading/current-edition/action-on-and-fn/#toc_the-on-modifier)

Pour terminer, Ember fournit également plusieurs events d'input spécifiques :  `@enter`, `@insert-newline` et `@escape-press`. Ces events n'existent pas sur les éléments input natifs, mais vous pourriez les trouver utiles pour implémenter les intéractions clavier.

> Le style Octane moderne pour gérer les events clavier est [d'écrire un modifier](../../upgrading/current-edition/glimmer-components/#toc_writing-your-own-modifiers) afin de séparer les responsabilités : le composant gère les états, tandis que le modifier gère les intéractions avec le DOM. L'action appelée lors d'un event reçoit alors un objet `event` en paramètre.

Reviewer: I'm not sure about 
> Le style Octane moderne pour gérer

What is modern, is the Octane style.

Maybe this is a little bit better : 

```suggestion
Le style moderne (ou Ember Octane) pour gérer les events clavier est [d'écrire un modifier](../../upgrading/current-edition/glimmer-components/#toc_writing-your-own-modifiers) afin de séparer les responsabilités : le composant gère les états, tandis que le modifier gère les intéractions avec le DOM. L'action appelée lors d'un event reçoit alors un objet `event` en paramètre.
```
